hooks/tk-nuke_actions.py

```diff
@@ -228,6 +230,57 @@ def _create_read_node(self, path, sg_publish_data):
             read_node["first"].setValue(seq_range[0])
             read_node["last"].setValue(seq_range[1])
 
+    def _sequence_range_from_path(self, path):
+        """
+        Parses the file name in an attempt to determine the first and last
+        frame number of a sequence. This assumes some sort of common convention
+        for the file names, where the frame number is an integer at the end of
+        the basename, just ahead of the file extension, such as
+        file.0001.jpg, or file_001.jpg. We also check for input file names with
+        abstracted frame number tokens, such as file.####.jpg, or file.%04d.jpg.
+
+        :param str path: The file path to parse.
+
+        :returns: None if no range could be determined, otherwise (min, max)
+        :rtype: tuple or None
+        """
+        # This pattern will match the following at the end of a string and
+        # retain the frame number or frame token as group(1) in the resulting
+        # match object:
+        #
+        # 0001
+        # ####
+        # %04d
+        #
+        # The number of digits or hashes does not matter; we match as many as
+        # exist.
+        frame_pattern = re.compile(r"([0-9#]+|[%]0\dd)$")
+        root, ext = os.path.splitext(path)
```

Comment:
Would it be better to split the path from the filename and only operate on the filename? This could help eliminate version number matches in the path itself. Maybe should include `.` separators before and after the frame number token? The `path_info` hook in `tk-multi-publish2` parses frame numbers this way. Not saying it's right for this case, but just for reference: https://github.com/shotgunsoftware/tk-multi-publish2/blob/master/hooks/path_info.py#L30

Reply:
Ah, I see the `$` anchor now. We can continue discussion of centralizing similar logic outside of this PR. No reason to block this. 👍